### PR TITLE
Add concurrency management

### PR DESCRIPTION
**Pull request by Dragnucs**

### CODE REVIEW

**Comment by Dragnucs** [comment]
Hello @madeindjs !

This PR uses the library Rayon to handle thread pools since it is well maintained, has recent commits and merged PRs, is old enough, and has a large user base. It also moves the code a bit more functional style to avoid bugs that went unnoticed like duplicate page fetching and parsing.

I stay available for any comments.

File: src/configuration.rs [hunk 1]
```diff
@@ -18,16 +18,17 @@ pub struct Configuration {
...
 
 impl Configuration {
     pub fn new() -> Self {
         Self {
-            respect_robots_txt: false,
-            verbose: false,
-            blacklist_url: Vec::new(),
-            user_agent: "spider/1.1.2",
-            delay: 0,
+            user_agent: "spider/1.2.1",
```
Comment (madeindjs) [inline src/configuration.rs:28]: Can you use 

```rs
const VERSION: &'static str = env!("CARGO_PKG_VERSION");
```

See [How can a Rust program access metadata from its Cargo package? - Stack Overflow](https://stackoverflow.com/questions/27840394/how-can-a-rust-program-access-metadata-from-its-cargo-package)

**Review by madeindjs — CHANGES_REQUESTED** [review]
Really great work! Thank you. I just emit a simple suggestion. Thank you

File: src/configuration.rs (hunk 1 above)
Comment (Dragnucs) [reply src/configuration.rs:28]: It turns out that the simplest tasks are the more difficult. I just don't know how to concatenate two strings.
I don't know what is happening, but string concatenation in Rust is more complicated than thread safety, parralelism and passing closures.

Comment (madeindjs) [reply src/configuration.rs:28]: Ok so I moved it in issue #11

**Pull request merged**